### PR TITLE
[BugFix] Fix NullPointerException in CatalogMgr.java

### PR DESCRIPTION
**Pull request by immutablecodes**
## Why I'm doing:

Fix NullPointerException in CatalogMgr.java

## What I'm doing:

Fix NullPointerException in CatalogMgr.java

## What type of PR is this:

- [x] BugFix
- [ ] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

Does this PR entail a change in behavior?

- [ ] Yes, this PR will result in a change in behavior.
- [x] No, this PR will not result in a change in behavior.

If yes, please specify the type of change:

- [ ] Interface/UI changes: syntax, type conversion, expression evaluation, display information
- [ ] Parameter changes: default values, similar parameters but with different default values
- [ ] Policy changes: use new policy to replace old one, functionality automatically enabled
- [ ] Feature removed
- [ ] Miscellaneous: upgrade & downgrade compatibility, etc.

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function
- [ ] This is a backport pr

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto-backported to the target branch
  - [x] 3.3
  - [x] 3.2
  - [x] 3.1
  - [ ] 3.0
  - [ ] 2.5

### CODE REVIEW

**Comment by alvin-celerdata** [comment]
@immutablecodes  can you sign-off this PR by `git commit -s`

**Comment by immutablecodes** [comment]
> @immutablecodes can you sign-off this PR by `git commit -s`

done

**Review by alvin-phoenix-ai — DISMISSED** [review]

File: fe/fe-core/src/main/java/com/starrocks/server/CatalogMgr.java [hunk 1]
```diff
@@ -191,7 +191,7 @@ public void alterCatalog(AlterCatalogStmt stmt) {
                 Map<String, String> properties = ((ModifyTablePropertiesClause) stmt.getAlterClause()).getProperties();
                 String serviceName = properties.get("ranger.plugin.hive.service.name");
 
-                if (serviceName.isEmpty()) {
+                if (serviceName == null || serviceName.isEmpty()) {
```
Comment (alvin-celerdata) [inline fe/fe-core/src/main/java/com/starrocks/server/CatalogMgr.java:194]:
```suggestion
                if (Strings.isNullOrEmpty(serviceName)) {
```

File: fe/fe-core/src/main/java/com/starrocks/server/CatalogMgr.java [hunk 2]
```diff
@@ -340,7 +340,7 @@ public void replayAlterCatalog(AlterCatalogLog log) {
             String catalogName = log.getCatalogName();
             Map<String, String> properties = log.getProperties();
             String serviceName = properties.get("ranger.plugin.hive.service.name");
-            if (serviceName.isEmpty()) {
+            if (serviceName == null || serviceName.isEmpty()) {
```
Comment (alvin-celerdata) [inline fe/fe-core/src/main/java/com/starrocks/server/CatalogMgr.java:343]:
```suggestion
            if (Strings.isNullOrEmpty(serviceName)) {
```

**Review by alvin-phoenix-ai — APPROVED** [review]

**Review by nshangyiming — APPROVED** [review]

**Pull request merged**